src/actions.js

```diff
@@ -0,0 +1,119 @@
+import {
+  graphql,
+  formatPageQueryWithCount,
+  formatMutation,
+  formatGQLString,
+  decodeId,
+  graphqlWithVariables,
+} from '@openimis/fe-core';
+import { ACTION_TYPE, MUTATION_SERVICE } from './reducer';
+import {
+  REQUEST, SUCCESS, ERROR, CLEAR,
+} from './utils/action-type';
+
+const TASK_GROUP_PROJECTION = () => [
+  'id',
+  'uuid',
+  'code',
+  'completionPolicy',
+  'taskexecutorSet { edges { node { user { id username lastName } } } }',
+];
+
+export const formatTaskGroupGQL = (taskGroup) => {
+  const executors = taskGroup?.executors?.map((executor) => decodeId(executor.id));
+  const executorsString = executors ? `[${executors.map((executorUuid) => `"${executorUuid}"`).join(', ')}]` : '[]';
+  const GQLString = `
+  ${taskGroup?.code ? `code: "${formatGQLString(taskGroup.code)}"` : ''}
+  ${taskGroup?.completionPolicy ? `completionPolicy: ${taskGroup.completionPolicy}` : ''}
+  ${taskGroup?.id ? `id: "${decodeId(taskGroup.id)}"` : ''}
+  ${taskGroup?.executors ? `userIds: ${executorsString}` : 'userIds: []'}
+  `;
+  return GQLString;
+};
+
+export function fetchTaskGroups(modulesManager, params) {
+  const payload = formatPageQueryWithCount('taskGroup', params, TASK_GROUP_PROJECTION());
+  return graphql(payload, ACTION_TYPE.SEARCH_TASK_GROUPS);
+}
+
+export function fetchTaskGroup(modulesManager, variables) {
+  return graphqlWithVariables(
+    `
+      query getTaskGroup ($taskGroupUuid: ID ) {
+        taskGroup(id: $taskGroupUuid) {
+          edges {
+            node {
+              id
+              uuid
+              code
+              completionPolicy
+              taskexecutorSet { edges { node { user { id username lastName } } } },
```

Comment:
I didn't want to use this projection, because I wanted to use _graphqlWithVariables_ and pass arguments as a variables instead. This ensures that quotation marks and backslashes are escaped properly. Anyway, I can use _formatPageQueryWithCount_ and projection if necessary.